js/shaders/scatter-fragment.glsl

```diff
@@ -33,39 +33,43 @@ vec2 rotate_xy(vec2 x, float angle) {
 
 void main(void) {
     vec2 pixel = (vUv - 0.5) * (pixel_size + 2.0 * stroke_width);
+    float fill_weight = 0.0;
 
 #if FAST_DRAW == FAST_CIRCLE
-    float r = length(pixel);
-    float fill_weight = 1.0 - smoothstep(pixel_size/2.0 - stroke_width - SMOOTH_PIXELS, pixel_size/2.0 - stroke_width + SMOOTH_PIXELS, r);
+    float r = length(pixel); // This uses `sqrt` which is slow, there might be room for performance improvements
+
+    fill_weight = 1.0 - smoothstep(pixel_size/2.0 - stroke_width - SMOOTH_PIXELS, pixel_size/2.0 - stroke_width + SMOOTH_PIXELS, r);
+
     float edge_weight = 1.0 - smoothstep(pixel_size/2.0 + stroke_width - SMOOTH_PIXELS, pixel_size/2.0 + stroke_width + SMOOTH_PIXELS, r);
     float stroke_weight = (1.0 - fill_weight) * edge_weight;
 
-    fill_weight *= (fill ? 1.0 : 0.0);
-
 #elif FAST_DRAW == FAST_SQUARE
-    float stroke_weight_inverse = smoothstep(-SMOOTH_PIXELS, SMOOTH_PIXELS, -(abs(pixel.x) - pixel_size/2.0 + stroke_width/2.0))
-                                * smoothstep(-SMOOTH_PIXELS, SMOOTH_PIXELS, -(abs(pixel.y) - pixel_size/2.0 + stroke_width/2.0));
-    float stroke_weight = 1.0 - stroke_weight_inverse;
-    float fill_weight =  1.0 * smoothstep(-SMOOTH_PIXELS, SMOOTH_PIXELS, -(abs(pixel.x) - pixel_size/2.0 + stroke_width/2.0))
-                             * smoothstep(-SMOOTH_PIXELS, SMOOTH_PIXELS, -(abs(pixel.y) - pixel_size/2.0 + stroke_width/2.0));
-    fill_weight *= (fill ? 1.0 : 0.0);
+    if (fill) {
```

Comment:
Did you benchmark this, I assumed a multiply to be faster than an if.

Reply:
I wanted to replace it with an `#if FILL`, but I changed my mind and put back the `fill_weight *= (fill ? 1.0 : 0.0);`, that's on the current status of the branch

Reply:
I guess `#if FILL` would not be a bad idea?

Reply:
I'm not sure why I did not do that. I think that is much better.